include/zephyr/drivers/flash.h

```diff
@@ -249,6 +256,27 @@ static inline int z_impl_flash_erase(const struct device *dev, off_t offset,
 	return rc;
 }
 
+#if defined(CONFIG_FLASH_RESET)
+/*
+ *  @brief  Send SPI reset command.
+ *
+ *  This api can be used  to reset the SPI flash memory i.e volatile
+ *  register configurations before SPI access.
+ *  Command also aborts any ongoing internal SPI operations.
+ *
+ *  @param  dev             : flash dev
+ */
+__syscall void flash_reset(const struct device *dev);
+
+static inline void z_impl_flash_reset(const struct device *dev)
+{
+	const struct flash_driver_api *api =
+		(const struct flash_driver_api *)dev->api;
+
+	api->send_flash_reset(dev);
```

Comment:
You've made this new API mandatory when `CONFIG_FLASH_RESET=y`, but you've only updated one driver.  There are likely boards that enable multiple flash driver types and this reset won't make sense for all type.s
```suggestion
	if (api->send_flash_reset == NULL) {
		return -ENOTSUP;
	}
	
	api->send_flash_reset(dev);
```

Reply:
Done